testdata/index-management/createSearchIndex.yml

```diff
@@ -26,7 +26,7 @@ tests:
       - name: createSearchIndex
         object: *collection0
         arguments:
-          model: { definition: &definition { mappings: { dynamic: true } } }
+          model: { definition: &definition { mappings: { dynamic: true } } , type: 'search' }
```

Comment:
These tests are failing because we do not include Type in `executeCreateSearchIndexes` and `executeCreateSearchIndex` operations. This can be fixed by updating the case [here](https://github.com/mongodb/mongo-go-driver/blob/9e50281d70f3b833c9431f2f71d51b3091bf50b6/mongo/integration/unified/collection_operation_execution.go#L325-L336) and [here](https://github.com/mongodb/mongo-go-driver/blob/9e50281d70f3b833c9431f2f71d51b3091bf50b6/mongo/integration/unified/collection_operation_execution.go#L369-L382) to include `Type`. e.g.

```go
var m struct {
	Definition interface{}
	Name       *string
	Type *string
}
err = bson.Unmarshal(val.Value, &m)
if err != nil {
	return nil, err
}
model := mongo.SearchIndexModel{
	Definition: m.Definition,
	Options:    options.SearchIndexes(),
}
model.Options.Name = m.Name
model.Options.Type = m.Type
models = append(models, model)
```

Reply:
Fixed